### ISSUE TITLE
Style/IfInsideElse autocorrect moves condition into the wrong place

### ISSUE THREAD

**Issue by denisdefreyne** [rubocop/rubocop#14821]
If an `else` block starts with a comment, `Style/IfInsideElse` should not move that comment down into the following `true` branch of the following condition.

--------

## Steps to reproduce the problem

Run the `Style/IfInsideElse` cop (with autoformatter) on the following:

```ruby
if cond_a
  body_a
else
  # Not A!
  
  if cond_b
    body_b
  else
    body_c
  end
end
```

## Expected behavior

No `Style/IfInsideElse` should be triggered, and the autoformatter should leave the code intact.

## Actual behavior

The code is autocorrected to the following

```ruby
if cond_a
  body_a
elsif cond_b
  # Not A!

  body_b
else
  body_c
end
```

Here, the comment `# Not A!` erroneously is inside the `cond_b` true condition.

```
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/denis/xxxxxxx/ma/blah.rb
C

Offenses:

blah.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
if cond_a
^
blah.rb:5:1: C: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
blah.rb:6:3: C: [Correctable] Style/IfInsideElse: Convert if nested inside else to elsif.
  if cond_b
  ^^

1 file inspected, 3 offenses detected, 3 offenses autocorrectable
Finished in 0.26967599987983704 seconds
```

## RuboCop version

```
1.82.1 (using Parser 3.3.10.0, Prism 1.8.0, rubocop-ast 1.49.0, analyzing as Ruby 3.3, running on ruby 3.3.6) [arm64-darwin25]
  - rubocop-factory_bot 2.28.0
  - rubocop-minitest 0.38.2
  - rubocop-performance 1.26.1
  - rubocop-rails 2.34.3
```

**Comment by hammadxcm** [comment]
Fix submitted in #14948. The cop now skips the offense when comments exist between the `else` keyword and the nested `if`, avoiding the incorrect autocorrection.

**Comment by hammadxcm** [comment]
> If an `else` block starts with a comment, `Style/IfInsideElse` should not move that comment down into the following `true` branch of the following condition.
> 
> ## Steps to reproduce the problem
> Run the `Style/IfInsideElse` cop (with autoformatter) on the following:
> 
> if cond_a
>   body_a
> else
>   # Not A!
>   
>   if cond_b
>     body_b
>   else
>     body_c
>   end
> end
> ## Expected behavior
> No `Style/IfInsideElse` should be triggered, and the autoformatter should leave the code intact.
> 
> ## Actual behavior
> The code is autocorrected to the following
> 
> if cond_a
>   body_a
> elsif cond_b
>   # Not A!
> 
>   body_b
> else
>   body_c
> end
> Here, the comment `# Not A!` erroneously is inside the `cond_b` true condition.
> 
> ```
> Use parallel by default.
> Skipping parallel inspection: only a single file needs inspection
> Inspecting 1 file
> Scanning /Users/denis/xxxxxxx/ma/blah.rb
> C
> 
> Offenses:
> 
> blah.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
> if cond_a
> ^
> blah.rb:5:1: C: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
> blah.rb:6:3: C: [Correctable] Style/IfInsideElse: Convert if nested inside else to elsif.
>   if cond_b
>   ^^
> 
> 1 file inspected, 3 offenses detected, 3 offenses autocorrectable
> Finished in 0.26967599987983704 seconds
> ```
> 
> ## RuboCop version
> ```
> 1.82.1 (using Parser 3.3.10.0, Prism 1.8.0, rubocop-ast 1.49.0, analyzing as Ruby 3.3, running on ruby 3.3.6) [arm64-darwin25]
>   - rubocop-factory_bot 2.28.0
>   - rubocop-minitest 0.38.2
>   - rubocop-performance 1.26.1
>   - rubocop-rails 2.34.3
> ```

Please confirm [denisdefreyne](https://github.com/denisdefreyne) the fix
